src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1909,6 +1909,45 @@ impl Cage {
         }
     }
 
+    //------------------------------------SYNC_FILE_RANGE SYSCALL------------------------------------
+
+     pub fn sync_file_range_syscall(&self, fd: i32, offset: isize, nbytes: isize, flags: u32) -> i32 {
+        let checkedfd = self.get_filedescriptor(fd).unwrap();
+        let mut unlocked_fd = checkedfd.write();
+        if let Some(filedesc_enum) = &mut *unlocked_fd {
+            match filedesc_enum {
+                File(ref mut normalfile_filedesc_obj) => {
+                    if is_rdonly(normalfile_filedesc_obj.flags) {
+                        return syscall_error(Errno::EBADF, "sync_file_range", "specified file not open for sync");
+                    }
+                    let inodeobj = FS_METADATA.inodetable.get(&normalfile_filedesc_obj.inode).unwrap();
+                    match &*inodeobj {
+                        Inode::File(_) => {
+                            let fobj = FILEOBJECTTABLE.get(&normalfile_filedesc_obj.inode).unwrap();
+                            let result = fobj.sync_file_range(offset, nbytes, flags).unwrap();
+                            match result {
```

Comment:
this seems wrong to do it like this, we should match result for ok or err instead of unwrapping it and wrap the errnos in err?

You should also be able to handle the libc errors with from discriminant (check out net calls for a few examples.